packages/bigquery/src/index.js

```diff
@@ -790,39 +790,42 @@ BigQuery.prototype.query = function(options, callback) {
 
   options = options || {};
 
-  if (options.params) {
-    options.useLegacySql = false;
-    options.parameterMode = is.array(options.params) ? 'positional' : 'named';
-
-    if (options.parameterMode === 'named') {
-      options.queryParameters = [];
-
-      for (var namedParamater in options.params) {
-        var value = options.params[namedParamater];
-        var queryParameter = BigQuery.valueToQueryParameter_(value);
-        queryParameter.name = namedParamater;
-        options.queryParameters.push(queryParameter);
-      }
-    } else {
-      options.queryParameters = options.params
-        .map(BigQuery.valueToQueryParameter_);
-    }
-
-    delete options.params;
-  }
-
   var job = options.job;
 
   var requestQuery = extend({}, options);
   delete requestQuery.job;
 
   if (job) {
     // Get results of the query.
+    delete requestQuery.params;
+    delete requestQuery.query;
+
     self.request({
```

Comment:
Just a style nitpick, but if we did an early return here it might make some of this a little more readable.